src/tests/ftest/pool/query_attribute.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Governments rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from apricot import TestWithServers
+from dmg_utils import get_pool_uuid_service_replicas_from_stdout
+from daos_utils import DaosCommand, get_pool_uuid, get_sizes
+
+
+class QueryAttributeTest(TestWithServers):
+    """Test class for daos pool query and attribute tests.
+
+    Test Class Description:
+        Query test: Create a pool and call daos pool query. From the output,
+        verify the UUID matches the one that was returned when creating the
+        pool. Also verify the pool size against the value that's passed in to
+        dmg pool create.
+
+        Attribute test:
+        1. Create 5 attributes to a pool. We can set attribute
+        name and corresponding value, so set unique name-value pair for each of
+        the 5 attributes.
+        2. Call pool list-attr. This returns the 5 attribute names, so compare
+        them against the actual names used.
+        3. Call pool get-attr. This returns the value of the corresponding
+        name, so do this for each of the 5 attributes.
+
+    :avocado: recursive
+    """
+
+    def test_query_attr(self):
+        """JIRA ID: DAOS-4624
+
+        Test Description:
+            Test daos pool query and attribute commands as described above.
+
+        Use Cases:
+            Test query, set-attr, list-attr, and get-attr commands.
+
+        :avocado: tags=all,pool,tiny,full_regression,pool_query_attr
+        """
+        # 1. Test pool query.
+        # Use the same format as pool query.
+        expected_size = "1000000000"
+        stdoutput = self.get_dmg_command().pool_create(
+            scm_size=expected_size).stdout
+        expected_uuid, service_replicas = \
+            get_pool_uuid_service_replicas_from_stdout(stdoutput)
```

Comment:
Mentioning this cause it's fresh in my mind since I just had to something similar for one of my tests. You can probably switch this to use the get_output() method. For example have this:
```
kwargs = {"scm_size": expected_size}
expected_uuid, service_replicas = self.daos_cmd.get_output("pool_create", **kwargs))
```

Then you'll have to add the regex in `dmg_utils.DmgCommand.METHOD_REGEX` dictionary which would be something like this:
`(?:UUID:|Service replicas:)\s+([A-Za-z0-9-]+)`

Then you can do something similar for the `daos_cmd.pool_query`

Reply:
Thanks for the regex. I made the change to use get_output.